src/cpp/session/modules/SessionDiagnostics.cpp

```diff
@@ -866,12 +867,43 @@ Error lintRSourceDocument(const json::JsonRpcRequest& request,
             documentId,
             isExplicit,
             isFragment);
-   
-   pResponse->setResult(lintAsJson(results.lint()));
-   
+
+   std::vector<module_context::SourceMarker> markers =
+       modules::markers::markersForFile(documentPath);
+   LintItems lintItems = results.lint();
+
+   using namespace module_context;
+   for (SourceMarker marker : markers)
+   {
+      LintType markerLintType;
+      switch (marker.type)
+      {
+      case SourceMarker::Type::Error:
+         markerLintType = LintTypeError;
+         break;
+      case SourceMarker::Type::Warning:
+         markerLintType = LintTypeWarning;
+         break;
+      case SourceMarker::Type::Info:
+         markerLintType = LintTypeInfo;
+         break;
+      case SourceMarker::Type::Style:
+         markerLintType = LintTypeStyle;
+         break;
+      default:
+         markerLintType = LintTypeInfo;
```

Comment:
Nit: minor line-count reduction
```suggestion
      case SourceMarker::Type::Style:
         markerLintType = LintTypeStyle;
         break;
      case SourceMarker::Type::Info:
      default:
         markerLintType = LintTypeInfo;
```